### DISCUSSION TITLE
How to get the type of a signal store ?

### DISCUSSION THREAD

**Discussion by t1m4lc** [ngrx/platform#4510] [Q&A]
I would like to have the type of my store to use it in a app initializer. 

for example i have 

```
export const UserStore = signalStore(
  { providedIn: 'root' },
  withDevtools('user'),
  withState(initialUserState),
  withMethods(store => {
  ...
  })
)
```

Typescript infer the type with something like :

```
UserStore: Type<{
    profile: DeepSignal<KeycloakProfile>;
    isLoggedIn: Signal<boolean>;
    loadUser: (payload: {
        profile: KeycloakProfile;
        roles: Role[];
    }) => Promise<...>;
} & StateSignal<...>>
```
But i would like something more condensed to it here for instance : 
```
const initializeUser = () => ({
  provide: APP_INITIALIZER,
  useFactory: (store: STORE_TYPE_HERE),
  multi: true,
  deps: [UserStore],
});
```
Anybody have some ideas to do it ? I try with `ReturType<typeof UserStore>` but it's not works...

**Comment by t1m4lc** [comment]
Workaround, i create 

```
interface PartialUserState {
  loadUser: (payload: {
    profile: KeycloakProfile;
    roles: Role[];
  }) => Promise<void>;
}

```
Because i only need to use loadUser in my intializer.

But this not resolve my question...

**Comment by brandonroberts** [comment]
Its in the FAQ for the SignalStore 
https://ngrx.io/guide/signals/faq
